pkg/driver/trace_driver_test.go

```diff
@@ -215,28 +217,48 @@ func TestDAGInvocation(t *testing.T) {
 		Runtime: 1000,
 		Memory:  128,
 	}
-	for i := 0; i < functionsToInvoke; i++ {
-		function = testDriver.Configuration.Functions[0]
-		list.PushBack(function)
+	functionList := make([]*common.Function, 3)
+	for i := 0; i < len(functionList); i++ {
+		functionList[i] = function
+	}
+	originalBranch := []*list.List{
+		func() *list.List {
+			l := list.New()
+			l.PushBack(&common.Node{Function: functionList[0], Depth: 0})
+			l.PushBack(&common.Node{Function: functionList[1], Depth: 1})
+			return l
+		}(),
+	}
+
+	newBranch := []*list.List{
+		func() *list.List {
+			l := list.New()
+			l.PushBack(&common.Node{Function: functionList[2], Depth: 1})
+			return l
+		}(),
 	}
 
+	rootFunction := originalBranch[0]
+	rootFunction.Front().Value.(*common.Node).Branches = newBranch
 	time.Sleep(2 * time.Second)
 
 	metadata := &InvocationMetadata{
-		RootFunction:        list,
+		RootFunction:        rootFunction,
 		Phase:               common.ExecutionPhase,
 		MinuteIndex:         0,
 		InvocationIndex:     2,
 		SuccessCount:        &successCount,
 		FailedCount:         &failureCount,
 		FailedCountByMinute: failureCountByMinute,
+		FunctionsInvoked:    &functionsInvoked,
 		RecordOutputChannel: invocationRecordOutputChannel,
 		AnnounceDoneWG:      announceDone,
 	}
 
 	announceDone.Add(1)
 	testDriver.invokeFunction(metadata)
-	if !(successCount == 1 && failureCount == 0) {
+	announceDone.Wait()
+	if !(functionsInvoked == 3 && failureCount == 0) {
```

Comment:
Consider adding the successCount test as well. I think it might be handled wrongly (see previous comment).